backend/windmill-queue/src/jobs.rs

```diff
@@ -589,6 +589,21 @@ pub async fn add_completed_job<
         )
         .await?;
     }
+    if queued_job.is_flow_step
+        && queued_job.job_kind != JobKind::Flow
+        && queued_job.job_kind != JobKind::FlowPreview
+        && queued_job.parent_job.is_some()
+    {
+        // persist the flow last progress timestamp to avoid zombie flow jobs
+        tracing::warn!("Persisting flow last progress timestamp");
+        sqlx::query!(
+            "UPDATE queue SET flow_last_progress_ts = now() WHERE id = $1 AND workspace_id = $2",
```

Comment:
need to be part of the pull otherwise we won't catch if the crash happen in between pull and here